### ISSUE TITLE
Member functions are invisible from IntelliSense at a direct call to the initializer list of a class

### ISSUE THREAD

**Issue by rohanbari** [microsoft/vscode-cpptools#7089]
**Type: IntelliSense**
In this following code segment (C++ IntelliSense for `c++20`):
```cpp
#include <array>
#include <iostream>

using std::array;
using std::cout;

class Initiator {
  int field;

public:
  constexpr Initiator(const int& param) { field = param; }
  constexpr uint32_t returnLength() const { return field; }
};

int main(void) {
  constexpr uint32_t size{Initiator{5}.returnLength()};
  array<uint32_t, size> myArray{3, 4, 5, 1, 9};

  for (auto&& it : myArray)
    cout << it << ' ';
  cout << std::endl;

  return 0;
}
```
Focus on this part of the first line of the main function:

    Initiator{5}.returnLength()

IntelliSense doesn't show access to any function used like this in a single line. **However, the code is completely alright and good.** For sake of clarity here's a screenshot:

![IntelliSense bug](https://user-images.githubusercontent.com/54104015/109629410-cb729a00-7b69-11eb-833f-c0b09681d1af.png)

- OS and Version: Linux Mint 20.1
- VS Code Version: Version: 1.53.2
- C/C++ Extension Version: v1.2.2
- Compiler: g++ (Ubuntu 10.2.0-5ubuntu1~20.04) 10.2.0
- Other extensions you installed (and if the issue persists after disabling them): Yes, I've installed MarkDown All-in-One, vscode-icons, Doxygen documentation generator, Prettier, and IntelliJ keybindings.
- Does this issue involve using SSH remotely to run the extension on a remote machine?: No
- Workspace is a single project (a single file).

---

**Here's the C/C++ extension diagnostic log:**

```json
-------- Diagnostics - 3/3/2021, 3:10:28 PM
Version: 1.2.2
Current Configuration:
{
    "name": "Linux",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [],
    "compilerPath": "/usr/bin/gcc",
    "cStandard": "c99",
    "cppStandard": "c++20",
    "intelliSenseMode": "linux-gcc-x64",
    "compilerArgs": [],
    "intelliSenseModeIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "compilerPathIsExplicit": true,
    "browse": {
        "path": [
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[ /code/main.cpp ]:
    /code/main.cpp
Translation Unit Configurations:
[ /code/main.cpp ]:
    Process ID: 4230
    Memory Usage: 88 MB
    Compiler Path: /usr/bin/gcc
    Includes:
        /usr/include/c++/10
        /usr/include/x86_64-linux-gnu/c++/10
        /usr/include/c++/10/backward
        /usr/lib/gcc/x86_64-linux-gnu/10/include
        /usr/local/include
        /usr/include/x86_64-linux-gnu
        /usr/include
    Standard Version: c++20
    IntelliSense Mode: linux-gcc-x64
    Other Flags:
        --g++
        --gnu_version=100200
Total Memory Usage: 88 MB

------- Workspace parsing diagnostics -------
Number of folders and files enumerated: 20355
Number of files discovered (not excluded): 18572
```

**Steps to reproduce**
1. Copy-paste the attached code.
2. Remove the code segment `returnLength()` and trigger suggest (<kbd>Ctrl</kbd> + <kbd>Space</kbd> in IntelliJ)
3. See the IntelliSense bug.

**Expected behavior**
I expect it to display the member functions specified in the class after the direct call to the initializer list.

**Comment by sean-mcmanus** [comment]
I'm not reproing the bug. Your screenshot shows an error on the line that defines "myArray" which is what occurs if C++17 is used. Are you using C++20? What compiler? C/C++: Log Diagnostics output could help.

![image](https://user-images.githubusercontent.com/19859882/109699894-f5b26080-7b45-11eb-9dbe-85d2b687842f.png)

**Comment by rohanbari** [comment]
@sean-mcmanus Yes, I used C++20 with g++10.2 compiler version. See the edit.

**Comment by sean-mcmanus** [comment]
Does your C/C++: Log Diagnostics output match mine, i.e. does it show g++-10 with c++20 is being used?

**Comment by rohanbari** [comment]
@sean-mcmanus Thank you for taking your time. You can now check the diagnostic log of **C/C++ extension** attached in the latest edit.

**Comment by sean-mcmanus** [comment]
Hmm, your logging appears like it should work. I'll see if anyone else on our team can repro this...

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. I wasn't reproing the bug previously because I wasn't deleting the returnLength when invoking completion. The bug also repros in VS. We should have a VS bug tracking this issue soon.

@bobbrow Not sure why you moved this issue to the "backlog".

**Comment by sean-mcmanus** [comment]
It's being tracked at https://developercommunity.visualstudio.com/t/GitHub-issue-7089---Member-functions-a/1367791

**Comment by rohanbari** [comment]
This issue has been fixed in Visual Studio 17.1.

**Comment by sean-mcmanus** [comment]
Yeah, looks like it got fixed a long time ago and we didn't notice.
